plugins/woocommerce/tests/php/includes/wc-order-functions-test.php

```diff
@@ -56,4 +56,59 @@ public function test_wc_restock_refunded_items_stock_metadata() {
 		$this->assertEquals( 0, (int) $line_item->get_meta( '_reduced_stock', true ) );
 		$this->assertEquals( 2, (int) $line_item->get_meta( '_restock_refunded_items', true ) );
 	}
+
+	/**
+	 * Test update_total_sales_counts and check total_sales after order reflection.
+	 *
+	 * Tests the fix for issue #23796
+	 */
+	public function test_wc_update_total_sales_counts() {
+
+		$product = WC_Helper_Product::create_simple_product();
+
+		WC()->cart->add_to_cart( $product->get_id() );
+
+		$order_id = WC_Checkout::instance()->create_order(
+			array(
+				'billing_email'  => 'a@b.com',
+				'payment_method' => 'dummy',
+			)
+		);
+
+		$this->assertEquals( 0, $product->get_total_sales() );
+
+		$order = new WC_Order( $order_id );
+
+		$order->update_status( 'processing' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		$order->update_status( 'cancelled' );
+		$this->assertEquals( 0, $product->get_total_sales() );
+
+		$order->update_status( 'processing' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		$order->update_status( 'completed' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		$order->update_status( 'refunded' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		$order->update_status( 'processing' );
+		$this->assertEquals( 1, $product->get_total_sales() );
+
+		// Test order trash / un-trash.
+		if ( $order->delete( false ) ) {
+			$this->assertEquals( 0, $product->get_total_sales() );
+
+			wp_untrash_post( $order->get_id() );
```

Comment:
This won't work when running the test suite with HPOS enabled, but I see we have a gap in our implementation—bear with me while I think about addressing that.

Reply:
With the latest changes in `trunk`, it is now possible to do `$order->untrash()` (datastore-agnostic untrashing, instead of relying on WP functions) which should allow us to make this test work even when HPOS is enabled.

Reply:
@barryhughes Great! thank you for letting me know.